Dolphin scripts/Entrance Randomizer/lib/graph_creation.py

```diff
@@ -0,0 +1,136 @@
+from __future__ import annotations
+
+from collections.abc import Mapping
+from pathlib import Path
+
+from lib.constants import *  # noqa: F403
+from lib.constants import __version__
+from lib.transition_infos import Area
+from lib.types_ import SeedType
+
+STARTING_AREA_COLOR = "#ff8000"  # Orange
+UPGRADE_AREAS_COLOR = "#0080ff"  # Blue
+STORY_TRIGGER_AREAS_COLOR = "#ff0000"  # Red
+UPGRADE_AREAS = {
+    LevelCRC.PLANE_COCKPIT,
+    LevelCRC.BITTENBINDERS_CAMP,
+    LevelCRC.MOUTH_OF_INTI,
+    LevelCRC.FLOODED_COURTYARD,
+    LevelCRC.TURTLE_MONUMENT,
+    LevelCRC.NATIVE_VILLAGE,
+    LevelCRC.RENEGADE_HEADQUARTERS,
+    LevelCRC.CAVERN_LAKE,
+    LevelCRC.MOUNTAIN_SLED_RUN,
+    LevelCRC.MOUNTAIN_OVERLOOK,
+    LevelCRC.APU_ILLAPU_SHRINE,
+}
+IMPORTANT_STORY_TRIGGER_AREAS = {
+    LevelCRC.ALTAR_OF_AGES,
+    LevelCRC.ST_CLAIRE_DAY,
+    LevelCRC.ST_CLAIRE_NIGHT,
+    LevelCRC.GATES_OF_EL_DORADO,
+}
+
+
+def create_vertices(
+    transitions_map: Mapping[tuple[int, int], tuple[int, int]],
+    starting_area: int,
+):
+    output_text = ""
+    areas_randomized: list[Area] = []
+    for original, redirect in transitions_map.items():
+        if TRANSITION_INFOS_DICT[original[0]] not in areas_randomized:
+            areas_randomized.append(TRANSITION_INFOS_DICT[original[0]])
+        if TRANSITION_INFOS_DICT[redirect[1]] not in areas_randomized:
+            areas_randomized.append(TRANSITION_INFOS_DICT[redirect[1]])
```

Comment:
This can be simplified using a set, which forces all elements to be unique.
```suggestion
    areas_randomized: set[Area] = set()
    for original, redirect in transitions_map.items():
        areas_randomized.add(TRANSITION_INFOS_DICT[original[0]])
        areas_randomized.add(TRANSITION_INFOS_DICT[redirect[1]])
```